python/py-upt/Portfile

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-upt
+version             0.6
+revision            0
+
+maintainers         {@korusuke somaiya.edu:karan.sheth} openmaintainer
+description         Package software from any package manager to any distribution
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            https://framagit.org/upt/upt
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            ${python.rootname}-${version}
+
+license             BSD-3-clause
+checksums           sha256  707b79dd699f5bdd68ef2eeb66bacfe2f86e0c93f99ad3a5c48b1b6a799b222b \
+                    rmd160  3b7b367dde8fe48315708de8cbbcf02034286fd5 \
+                    size    23609
+
+python.versions     37
+
+if {${name} ne ${subport}} {
+
+    depends_lib-append \
+                        port:py${python.version}-spdx-lookup \
+                        port:py${python.version}-setuptools
+
+    livecheck.type      none
+
+
+    post-destroot {
+        set docdir ${prefix}/share/doc/${subport}
+        xinstall -d ${destroot}${docdir}
+        xinstall -m 0644 -W ${worksrcpath} README.md LICENSE \
+            ${destroot}${docdir}
+    }
+
+    default_variants    +macports +cpan +pypi
```

Comment:
You wanted my opinion about variants: I would get rid of them and simply always include everything.

Homebrew threw away all of their variants from the core packages. We need variants when adding some huge dependencies like Qt, or when one needs to build the port in a different way to pick between X11 and native GUI. These three dependencies are soooooo tiny that it makes absolutely no sense to add the extra level of complexity on top of the `upt` port. Sure, someone who wants to turn a cpan package into a macports port might not need to install pypi, but will that person really care about not installing a few additional text files?

One issue (mental exercise) with dependencies I had last time was that once you install `py-upt-cpan`, this is in fact completely useless without `upt`, so it made perfect sense to make `py-upt-cpan` depend on `upt`. Then again, when I install `upt`, I do want all the backends and all the frontends installed automatically, I don't want to manually install 6 different packages to run one command. So it wasn't even remotely clear to me in which direction dependencies should go. Does `py-upt-cpan` depend on `py-upt`, or is it vice versa? In any case we may not introduce circular dependencies. I assume that dependencies as declared right now are ok (even if somewhat useless to install `py-upt-cpan` alone), but I would not introduce variants, at least not until we add a frontend that requires some heavy dependencies to be installed.

(The reason why I at some point suggested to create both `upt` and `py-upt` was that `py-upt-cpan` would depend on `py-upt`, while `upt` would depend on all the backends and all the frontends. But that would be more of a meta port, and the current layout as it is in this PR does the job as well without the extra complication of that metaport.)

Reply:
So should we get rid of variants and by default install all frontends and only macports backend by default when installing`py-upt` ?

Reply:
sounds good to me!